petroflow/src/well_segment.py

```diff
@@ -2063,34 +2068,48 @@ def equalize_histogram(self, src=None, dst=None, channels='last'):
             setattr(self, _dst, img)
         return self
 
-    def shift_logs(self, max_period, mnemonics=None):
-        """Shift every `logs` column from `mnemonics` by a random step sampled
-        from discrete uniform distribution in [-`max_period`, `max_period`].
-        All new resulting empty positions are filled with first/last
-        column value depending on shift direction.
+    def random_shift_logs(self, max_shift, src=None):
+        """Shift `logs` attr columns by a step sampled from discrete uniform
+        distribution in [-`max_period`, `max_period`]. Resulted empty positions
+        are filled with first/last column value depending on shift direction.
 
         Parameters
         ----------
-        max_period : positive int or str
+        max_shift : positive int or str
             Maximum possible shift in centimeters. If `str`, must be specified
             in a <value><units> format (e.g. "10m").
-        mnemonics : None or str or list of str
-            - If `None`, shift all logs columns.
-            - If `str`, shift single column from logs with `mnemonics` name.
-            - If `list`, shift all logs columnns with names in `mnemonics`.
-            Defaults to `None`.
 
         Returns
         -------
         self : type(self)
-            Self with shifted logs columns.
+            Self with randomly shifted `logs` columns.
         """
-        max_period = parse_depth(max_period, check_positive=True, var_name="max_period")
-        mnemonics = self.logs.columns if mnemonics is None else to_list(mnemonics)
-        max_period = int(max_period / self.logs_step)
-        periods = np.random.randint(-max_period, max_period + 1, len(mnemonics))
-        for mnemonic, period in zip(mnemonics, periods):
-            fill_index = 0 if period > 0 else -1
-            fill_value = self.logs[mnemonic].iloc[fill_index]
-            self.logs[mnemonic] = self.logs[mnemonic].shift(periods=period, fill_value=fill_value)
+        df = self.logs
+        src = df.columns if src is None else [src] if isinstance(src, str) else src
+        max_shift = parse_depth(max_shift, check_positive=True, var_name="max_shift") // self.logs_step
+        if max_shift == 0:
+            warnings.warn('Passed `max_shift` is smaller then dataframe index step and therefore no shift was applied.')
+        for column, series in df[src].iteritems():
+            periods = np.random.randint(-max_shift, max_shift + 1)
+            fill_value = series.iloc[0] if periods > 0 else series.iloc[-1]
+            df[column] = series.shift(periods=periods, fill_value=fill_value)
         return self
+
+    @process_columns(dst_from_result_=True)
+    def one_hot_encode(self, df, encoder):
+        """One-hot encode columns of WellSegment attribute.
+
+        Parameters
+        ----------
+        encoder : callable
```

Comment:
```suggestion
        encoder : sklearn.preprocessing.OneHotEncoder
```
I'd explicitly define type here while describing, that it should be fitted on the specified `src` columns from `attr` in the next line.